libmattermost.c

```diff
@@ -2298,7 +2326,10 @@ mm_process_msg(MattermostAccount *ma, JsonNode *element_node)
 				}
 			}
 
-			if (!purple_strequal(channel_id,"")) {
+			const gint64 since = mm_get_channel_approximate_view_time(ma, channel_id);
+
+			// If a new channel is joined then first show the last 60 messages and then channel joined message
+			if (!purple_strequal(channel_id, "") && since != MATTERMOST_NEW_CHANNEL_FOUND) {

```

Comment:
Are you using value of since somewhere else. If not then you can apply my suggestion.
```suggestion
			// If a new channel is joined then first show the last 60 messages and then channel joined message
			if (!purple_strequal(channel_id, "") && mm_get_channel_approximate_view_time(ma, channel_id) != MATTERMOST_NEW_CHANNEL_FOUND) {

```